extensions/interactions/ImageClickInput/ImageClickInput.py

```diff
@@ -68,3 +70,227 @@ class ImageClickInput(base.BaseInteraction):
+        """Creates a RuleSpec proto object list.
+
+        Args:
+            rule_specs_list: list(RuleSpec). List of rule specifications.
+
+        Returns:
+            list. The RuleSpec proto object list.
+        """
+        rule_specs_list_proto = []
+        rules_specs_proto = {}
+
+        for rule_spec in rule_specs_list:
+            rule_type = rule_spec.rule_type
+            if rule_type == 'IsInRegion':
+                is_in_region_proto = cls._to_is_in_image_region_proto(
+                    rule_spec.inputs['x'])
+                rules_specs_proto = state_pb2.ImageClickInputInstance.RuleSpec(
+                    is_in_region=is_in_region_proto)
+
+            rule_specs_list_proto.append(rules_specs_proto)
+
+        return rule_specs_list_proto
+
+    @classmethod
+    def _to_is_in_image_region_proto(cls, is_in_region):
+        """Creates a IsInRegionSpec proto object.
+
+        Args:
+            is_in_region: str. The input region name.
+
+        Returns:
+            IsInRegionSpec. The IsInRegionSpec proto object.
+        """
+        is_in_region_proto = (
+            state_pb2.ImageClickInputInstance.RuleSpec.IsInRegionSpec(
+                input_region=is_in_region))
+
+        return is_in_region_proto
+
+    @classmethod
+    def _to_customization_args_proto(cls, customization_args):
+        """Creates a CustomizationArgs proto object
+            for ImageClickInputInstance.
+
+        Args:
+            customization_args: dict. The customization dict. The keys are
+                names of customization_args and the values are dicts with a
+                single key, 'value', whose corresponding value is the value of
+                the customization arg.
+
+        Returns:
+            CustomizationArgs. The CustomizationArgs proto object.
+        """
+        image_and_regions_proto = cls._to_image_and_regions_proto(
+            customization_args['imageAndRegions'])
+
+        customization_arg_proto = (
+            state_pb2.ImageClickInputInstance.CustomizationArgs(
+                image_and_regions=image_and_regions_proto))
+
+        return customization_arg_proto
+
+    @classmethod
+    def _to_image_and_regions_proto(cls, imageAndRegionsList):
+        """Creates a ImageWithRegions proto object.
+
+        Args:
+            imageAndRegionsList: list. The list of
+                image and regions.
+
+        Returns:
+            ImageWithRegions. The ImageWithRegions proto object.
+        """
+        image_file_path = imageAndRegionsList.value['imagePath']
+
+        labeled_regions_list_proto = cls._to_labeled_region_list_proto(
+            imageAndRegionsList.value['labeledRegions'])
+
+        image_with_regions_proto = objects_pb2.ImageWithRegions(
+            image_file_path=image_file_path,
+            labeled_regions=labeled_regions_list_proto)
+
+        return image_with_regions_proto
+
+    @classmethod
+    def _to_labeled_region_list_proto(cls, labeledRegionsList):
```

Comment:
```suggestion
    def _to_labeled_region_list_proto(cls, labeled_regions_list):
```
Why wasn't this caught by linter? cc @DubeySandeep

Reply:
Done